presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergAbstractMetadata.java

```diff
@@ -498,7 +506,15 @@ public TableStatisticsMetadata getStatisticsCollectionMetadata(ConnectorSession
 
     private List<ColumnStatisticMetadata> getColumnStatisticMetadata(ConnectorSession session, ColumnMetadata col)
     {
-        return MetastoreUtil.getSupportedColumnStatistics(col.getType()).stream().map(x -> new ColumnStatisticMetadata(col.getName(), x)).collect(Collectors.toList());
+        return MetastoreUtil.getSupportedColumnStatistics(col.getType()).stream().map(x ->
+        {
+            if (this.useSampleForAnalyze && useSampleStatistics(session) && x.name().equals("NUMBER_OF_DISTINCT_VALUES")) {
```

Comment:
instead of `x.name().equals()` you can just use `x.equals(ColumnStatisticType.NUMBER_OF_DISTINCT_VALUES)`